### PR TITLE
update the proteinpaint-client version to 2.50.0

### PR DESCRIPTION
**Pull request by siosonel**
## Description

Also,  handle empty userDetails.data in sequence read app

Features:
- include auth test status in server healthcheck
- Hide synonymous mutations by default for GDC oncoMatrix
- Improve the matrix sorting options to easily toggle sorting by cnv and/or consequence

Fixes:
- fix the error from genomic alterations rendering when there are no mutations or CNV data
- In GDC query, do not supply empty "case_filters{content[]}" that will slow down API. lollipop and oncomatrix are now faster when there's no cohort
- at GDC bam slicing UI, the table listing available cases and bam files can be filtered by assay types
- Add to GDC oncoMatrix mutation/cnv buttons all available mutation/cnv classes in GDC instead of all available mutation/cnv classes in the current matrix

## Checklist

- [ ] Added proper unit tests
- [ ] Left proper TODO messages for any remaining tasks
- [ ] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

### CODE REVIEW

File: packages/portal-proto/src/features/proteinpaint/SequenceReadWrapper.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ export const SequenceReadWrapper: FC<PpProps> = (props: PpProps) => {
   useEffect(
```
Comment (craigrbarnes) [inline packages/portal-proto/src/features/proteinpaint/SequenceReadWrapper.tsx]: we have been using `useDeepCompareEffect`, especially when the dependencies are objects. I suggest using it here. The library's homepage is: https://github.com/sandiiarov/use-deep-compare#readme

**Review by craigrbarnes — CHANGES_REQUESTED** [review]
See my comment about `useDeepCompareEffect`.

File: packages/portal-proto/src/features/proteinpaint/SequenceReadWrapper.tsx (hunk 1 above)
Comment (siosonel) [reply packages/portal-proto/src/features/proteinpaint/SequenceReadWrapper.tsx]: Ok, thanks, I've replaced with `useDeepCompareEffect()`.

Comment (craigrbarnes) [reply packages/portal-proto/src/features/proteinpaint/SequenceReadWrapper.tsx]: Thank you. I will merge it now

**Review by craigrbarnes — APPROVED** [review]

**Pull request merged**